### ISSUE TITLE
newton does not bound roots

### ISSUE THREAD

**Issue by dpsanders** [dpsanders/ValidatedNumerics.jl#164]
Compare the difference between using `Float64` and `BigFloat`:

```
function iterate(f, n, x)
    for i in 1:n
        x = f(x)
    end
    return x
end

f(x) = 4x * (1-x) 

julia> newton(x->iterate(f, 5, x) - x, 0..1)
17-element Array{ValidatedNumerics.RootFinding.Root{Float64},1}:
 ValidatedNumerics.RootFinding.Root{Float64}([0, 1.6768e-16],:unknown)        
 ValidatedNumerics.RootFinding.Root{Float64}([0.00903565, 0.00903566],:unique)
 ValidatedNumerics.RootFinding.Root{Float64}([0.010235, 0.0102351],:unique)   
 ValidatedNumerics.RootFinding.Root{Float64}([0.035816, 0.0358161],:unique)   
 ValidatedNumerics.RootFinding.Root{Float64}([0.040521, 0.0405211],:unique)   
 ValidatedNumerics.RootFinding.Root{Float64}([0.0793732, 0.0793733],:unique)  
 ValidatedNumerics.RootFinding.Root{Float64}([0.0896182, 0.0896183],:unique)  
 ValidatedNumerics.RootFinding.Root{Float64}([0.138132, 0.138133],:unique)    
 ValidatedNumerics.RootFinding.Root{Float64}([0.155516, 0.155517],:unique)    
 ValidatedNumerics.RootFinding.Root{Float64}([0.209971, 0.209972],:unique)    
 ValidatedNumerics.RootFinding.Root{Float64}([0.235517, 0.235518],:unique)    
 ValidatedNumerics.RootFinding.Root{Float64}([0.292292, 0.292293],:unique)    
 ValidatedNumerics.RootFinding.Root{Float64}([0.326347, 0.326348],:unique)    
 ValidatedNumerics.RootFinding.Root{Float64}([0.38212, 0.382121],:unique)     
 ValidatedNumerics.RootFinding.Root{Float64}([0.424286, 0.424287],:unique)    
 ValidatedNumerics.RootFinding.Root{Float64}([0.476209, 0.47621],:unique)     
 ValidatedNumerics.RootFinding.Root{Float64}([0.75, 0.75],:unknown)           

julia> newton(x->iterate(f, 5, x) - x, @biginterval(0, 1))
33-element Array{ValidatedNumerics.RootFinding.Root{BigFloat},1}:
 ValidatedNumerics.RootFinding.Root{BigFloat}([0, 3.51649e-10]₂₅₆,:unknown)       
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.00903565, 0.00903566]₂₅₆,:unique)
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.010235, 0.0102351]₂₅₆,:unique)   
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.035816, 0.0358161]₂₅₆,:unique)   
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.040521, 0.0405211]₂₅₆,:unique)   
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.0793732, 0.0793733]₂₅₆,:unique)  
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.0896182, 0.0896183]₂₅₆,:unique)  
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.138132, 0.138133]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.155516, 0.155517]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.209971, 0.209972]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.235517, 0.235518]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.292292, 0.292293]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.326347, 0.326348]₂₅₆,:unique)    
 ⋮                                                                                
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.749999, 0.750001]₂₅₆,:unknown)   
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.75, 0.750001]₂₅₆,:unknown)       
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.806052, 0.806053]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.82743, 0.827431]₂₅₆,:unique)     
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.879379, 0.87938]₂₅₆,:unique)     
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.893026, 0.893027]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.937173, 0.937174]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.944417, 0.944418]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.977069, 0.97707]₂₅₆,:unique)     
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.979746, 0.979747]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.997434, 0.997435]₂₅₆,:unique)    
 ValidatedNumerics.RootFinding.Root{BigFloat}([0.997735, 0.997736]₂₅₆,:unique)    
```

**Comment by dpsanders** [comment]
This seems to be very sensitive to the exact interval in which roots are searched for.

**Comment by dpsanders** [comment]
Seems to be fixed by #254.

**Comment by lbenet** [comment]
While the solution in #254 seems consistent, does it really solve this issue? 

Regarding the [first example](https://github.com/dpsanders/ValidatedNumerics.jl/issues/164#issue-168414835), I'm getting 32 roots when using `Interval{Float64}`, while 33 for `Interval{BigFloat}`. It seems that, for `BigFloat`s the root around 0.75 is the tricky one...

**Comment by dpsanders** [comment]
The problem was that previously the `Float64` version just ignored the roots in [0.75, 1].
The issue with `BigFloat`s can easily be fixed by increasing the depth:

```
julia> roots = newton(x->iterate(f, 5, x) - x, @biginterval(0, 1), maxlevel=50); 
julia> length(roots)
32
```

The correct number is 2^n, where n is the order of iteration, due to the exact mapping to the tent map.

So #254 does fix the issue with Newton, as far as I can see (but currently introduces a problem with Krawczyk that I don't understand).

**Comment by dpsanders** [comment]
Fixed in #254.
